GapLocker/PluginInstance.h

```diff
@@ -144,11 +149,13 @@ class CPluginInstance : public pluginbase::Mt5Plugin,
         {
             return (retcode);
         }
-
-        bool isReadedAllParams = SettingsReader::Read(serverApi, pluginSettings);
-        if (!isReadedAllParams)
+        
         {
-            return (MT_RET_ERR_PARAMS);
+            LOCK();
+            if (!SettingsReader::Read(serverApi, pluginSettings))
```

Comment:
Тут чтобы не ставить лишние скобки, можно сделать такой финт:
```suggestion
            if (LOCK(); !SettingsReader::Read(serverApi, pluginSettings))
```